### ISSUE TITLE
Upload doesn't seem to be updating JS or CSS files

### ISSUE THREAD

**Issue by scferg** [driftyco/ionic-view-issues#10]
After I do `ionic upload`, everything seems to go well in the CLI, but once in the Ionic View app, the only things that seem to get updated in the project are the HTML files and not the JS or CSS.

Everything renders fine and without errors in the browser, but I can't seem to get those other files updated in the View app.

**Comment by tlancina** [comment]
are your css and js files not in www?

**Comment by scferg** [comment]
They sure are. They're in the default locations, which is why I don't understand why they're not being changed in the View app.

Even adding a simple `alert` to make it super obvious that's it's changed doesn't come over. It does in the browser, though.

**Comment by tlancina** [comment]
Hmm, well all the cli is doing is zipping your www directory and shipping
it off. So it's weird that HTML changes are there but not the others...

Are you syncing to the latest version before viewing the app in View?

**Comment by scferg** [comment]
Yeah I am syncing to the latest version (which says it syncs successfully). I might try grabbing the zip file real quick before its done uploading and gets deleted and see what it looks like.

If I could delete the app from View, I could try uploading a fresh copy. Would removing the app ID in the config file basically accomplish the same thing?

**Comment by tlancina** [comment]
Normally I would advise against touching the app_id, as it's not meant to
be user-configured and we'll be moving it out of ionic.project.

However in this instance yes, if you set it to a blank string it will show
up as a new app in View with a new ID.

If you changed the name of the CSS file before uploading does that have any
effect? I'm wondering if it's a weird caching issue.

**Comment by scferg** [comment]
Yep, renaming the JS and CSS files does the trick! Must be a caching issue...

What do you advise going forward?

Thanks for the help, by the way.

**Comment by tlancina** [comment]
Not a problem, if you had this issue I'm sure others will as well. Thanks for bringing it to our attention!

I think I know where the problem is with inAppBrowser, so I'll try and have this out with the upcoming update to let people delete apps, since getting cached files kind of defeats a lot of the purpose of View.

See if clearing the app data within View and then re-syncing helps. Other than that, not much I can recommend for you at the moment other than changing the filename to avoid the cache.

**Comment by tlancina** [comment]
Just an update - this is apparently unfixable with UIWebView.  It seems that no matter what cache policy you set, it will only apply to the initial request (in this case index.html) and load all CSS and JS from the cache no matter what.  There are many, many StackOverflow posts regarding this issue and the only thing that seems to work is adding a random string to the url of any assets your index.html file loads.

http://stackoverflow.com/questions/2523435/how-to-clear-uiwebview-cache

Currently brainstorming creative solutions, but I think what will end up happening is we add a hook to the ionic-cli that adds the cachebuster to asset urls in index.html before it gets zipped and uploaded.

**Comment by mlynch** [comment]
Ionic deploy should fix this as well once that is ready

On Wed, Feb 4, 2015 at 3:26 PM, Tim Lancina <notifications@github.com>
wrote:

> Just an update - this is apparently unfixable with UIWebView. It seems
> that no matter what cache policy you set, it will only apply to the initial
> request (in this case index.html) and load all CSS and JS from the cache no
> matter what. There are many, many StackOverflow posts regarding this issue
> and the only thing that seems to work is adding a random string to the url
> of any assets your index.html file loads.
>
> http://stackoverflow.com/questions/2523435/how-to-clear-uiwebview-cache
>
> Currently brainstorming creative solutions, but I think what will end up
> happening is we add a hook to the ionic-cli that adds the cachebuster to
> asset urls in index.html before it gets zipped and uploaded.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/driftyco/ionic-view-issues/issues/10#issuecomment-72942791>
> .
>



-- 
Max Lynch
CEO and Co-founder
http://ionicframework.com/

**Comment by numerized** [comment]
Ok I'm adding a +1 on this, the only solution for me is to remove the ionic.project app.id.
Clearing all cache doesn't do the trick.
Big thanks for this amazing tool.

**Comment by RomanMinkin** [comment]
Happens for me as well, solves with clearing app data and restarting IonicView app (double tap on home button and closing process)

**Comment by mnewmedia** [comment]
lol wastet so much time, why didn't i look at the issues earlier.. =)

**Comment by Alexintosh** [comment]
Android: For me as well, solves with clearing app data and restarting IonicView app as RomanMinkin.

**Comment by fi5u** [comment]
On iOS, at least for me, I don't need to clear app data. Just quitting Ionic View and reopening updates the app with the latest assets.

**Comment by dani-inside** [comment]
+1 also happening on iOS8 – the tool itself is great, I'm looking forward to the new functions like sharing with customers.

**Comment by isherwood** [comment]
fi5u's solution resolve hours of pain for me. For those not familiar with iOS, you want to double-click the home button, swipe horizontally to Ionic View, then swipe it upward to end the process. After days of frustration and avoidance, my app is finally updating. Thank you.

**Comment by sanderboom** [comment]
+1 
Geat app!

Ionic View is unfortunately not updating the app at all. 
The only solution working for me is removing `app_id` in `ionic.project`

Loading on iOS8.3
```
ionic info
Your system information:
OS: Distributor ID:     ManjaroLinux Description:       Manjaro Linux 
Node Version: v1.6.3
Cordova CLI: 4.3.0
Ionic CLI Version: 1.3.19
```

**Comment by aliabbas** [comment]
Tried everything like @sanderboom, wasn't updating the application.  

Clearing 'app_id' in ionic.project did the trick

$ ionic info
Your system information:
OS: Mac OS X Yosemite
Node Version: v0.12.0
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.3.19
Xcode version: Xcode 6.3 Build version 6D570 
ios-sim version: 3.1.1 
ios-deploy version: 1.5.0

**Comment by Bret-Finley** [comment]
Thanks @aliabbas, developing on android and nothing was working, looks like clearing out the app_id worked liked a charm

**Comment by adamduren** [comment]
I am experiencing this problem as well. Clearing out the app id works but it generates a new project in the Ionic Apps view which is not desirable.

**Comment by Judavaro** [comment]
Hi guys,

Also I have the problem , the workaround is to delete app_id and " ionic charge " again. The system creates a new application with a diferent app_id . So What if we have many deployments?

Tks,

**Comment by ridan** [comment]
If I try to delete the app_id in the ionic.project file it gives me an error saying "App XXXXXXX not found Unable to upload app (CLI v1.3.19)". The app on the ionic-view is always the first one.

**Comment by TvBrienen** [comment]
@ridan You should only clear the app_id itself, and leave "app_id": "" in there. Ionic will then generate a new app_id for you.

**Comment by adriantombu** [comment]
So, 3 months later, no fix at all. That's sad because it is very useful :/

**Comment by gigocabrera** [comment]
I cleared the app_id itself and left "app_id": "" in there but I got an error message and the app was not uploaded. I also tried the option from @fi5u to no avail. I'm thinking I may create a new account, rename the app and try that way, maybe???

**Comment by f1lander** [comment]
I have the same problem, is there any solution without clear the app_id every time ?

**Comment by tlancina** [comment]
Hey all, just a quick update - after the recent flurry of activity on this issue (and lots of reports on twitter) I took a look and it seems that upload is currently broken either on the server or from changes to the cli.  We're looking into it now!

**Comment by tlancina** [comment]
Backend issues have been resolved, there is a chance that old apps will still have the updating issue.  If this is the case, try changing app_id to force a new project to be created.

Regarding the actual original issue, I've just added a hook to the cli which should fix it in the next ionic-cli release.

**Comment by adriantombu** [comment]
Great, will test that tomorrow !
Thanks a lot :)

**Comment by Judavaro** [comment]
Thks i'm go to test

**Comment by sanderboom** [comment]
Tested and seems to work. Thanks!

**Comment by gigocabrera** [comment]
Yes, it worked for me too on my second try after change the app_id. 
Thank you @tlancina, you're awesome!!!

**Comment by adriantombu** [comment]
Works great with me too, you're the man !

**Comment by ckahle33** [comment]
What is the best / safest way to change the `app_id` Thanks. Appreciate the fix for this!

**Comment by tlancina** [comment]
@ckahle33 easiest way is to just edit your ionic.project file and change `app_id` to "", then run `ionic upload`.

**Comment by ckahle33** [comment]
@tlancina - Many thanks for all of the hard work. Cheers.

**Comment by AlexKvazos** [comment]
Still no updates? I am getting old cached assets.

**Comment by Judavaro** [comment]
No runin,

2015-04-23 16:01 GMT-05:00 Alejandro Cavazos <notifications@github.com>:

> Still no updates? I am getting old cached assets.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/driftyco/ionic-view-issues/issues/10#issuecomment-95717522>
> .
>



-- 
Coridialmente,

Juan David Vanegas Romero
Técnico en programación de sistemas de información
PSP Fundamentals
ISTQB Fundamentals
3013506792

**Comment by tlancina** [comment]
Could someone upgrade to the latest ionic-cli (1.3.22) and tell me if this is fixed for them?

**Comment by Judavaro** [comment]
Ok,i going to test it

**Comment by MCollins434** [comment]
Experienced the "Blank app on Android" issue even using the io starter template with no additions except my appID.
Updated to 1.3.22 just now and everything's looking good!!

Thank you!!!!

**Comment by c0bra** [comment]
Still getting cached apps here.

**Comment by tlancina** [comment]
@c0bra ios or android?

**Comment by c0bra** [comment]
Android.  I deleted the app id from `ionic.project` as noted above, uploaded again, used the newly generated app id and got the new version.

**Comment by m0rganic** [comment]
App update works but it does not persist through app restarts on iOS and Android. 

If we restart the app after running through the manual ionic deploy update process we get the old version and the update seems to not have taken place.

**Comment by tlancina** [comment]
@m0rganic are you sure? Forgive my skepticism, but at least on iOS (and I believe Android as well) the files are overwritten, so if you see the update once it means that the files have updated successfully and should be purged from memory once you close the app.

**Comment by m0rganic** [comment]
@tlancina I'm definitely still seeing the issue. Perhaps I don't fully understand the solution suggested with blanking out the app_id in ionic.project.

```
$ionic info

Your system information:

OS: Mac OS X Yosemite
Node Version: v0.10.38
Cordova CLI: 5.0.0
Ionic Version: 1.0.0-rc.5
Ionic CLI Version: 1.3.22
Xcode version: Xcode 6.3.1 Build version 6D1002
ios-sim version: 3.1.1
ios-deploy version: 1.5.0

```

Perhaps someone who know's how to make this work can modify the example [ionic-starter-deploy](https://github.com/driftyco/ionic-starter-deploy)?

**Comment by tlancina** [comment]
@m0rganic are trying to test the deploy plugin within View? View functions essentially the same as the deploy plugin, by downloading your latest files and loading them.  This issue is relating to assets remaining cached even after you hit Sync and then view the app.

If you are having issues with deploy on View you should open a different issue :)

**Comment by m0rganic** [comment]
@tlancina my apologies I'm fairly new to ionic and assumed View was a feature that the deploy service utilized. I'll post a new issue to the ionic deploy service project.

**Comment by ridan** [comment]
@TvBrienen thanks, worked like a charm

**Comment by gigocabrera** [comment]
@tlancina I've noticed something strange every time I upload my app. It started happening recently. It adds the characters below to my index.html page right before the doctype. The app works fine but it adds an extra empty line when I run the app in the View

&#xFEFF;<!DOCTYPE html>

**Comment by stalepettersen** [comment]
I'm experiencing the same issue, JS and CSS are cache or not loading (how can I debug this?).

Renaming `www/css/style.css` to `www/css/style2.css` and updating reference in index.html does not work. Setting `"app_id":""` in `ionic.project` has no effect. Inlining the CSS in HTML makes the CSS changes visible.. but not a viable solution I think.

```
OS: Mac OS X Yosemite
Node Version: v0.10.28
Cordova CLI: 5.0.0
Ionic Version: 1.0.0-rc.5
Ionic CLI Version: 1.3.21
Xcode version: Xcode 5.0.2 Build version 5A3005
ios-sim version: Not installed
ios-deploy version: Not installed
```

Running latest Ionic View on Android 5.0.2 device.

**Comment by schris12** [comment]
This issue is still on for me on android but it seems that when I clear the Ionic View app data and reopen the app it works ok

**Comment by gigocabrera** [comment]
I was able to confirm that the issue with the extra characters being added to my index.html page only happens to one of my apps and not the other one. 

    &#xFEFF;<!DOCTYPE html>

I'm using 
OS: Windows 8.1
Node Version: v0.12.2
Cordova CLI: 4.3.0
Ionic Version 1.0.0-beta.14
Ionic CLI Version: 1.3.22

Any ideas? How can I troubleshoot?

**Comment by Avareto** [comment]
I still have the problem.
I updated to cli 1.3.22, terminated my App, cleared App date and re-download the app.
Before, I removed all history and website data from Safari settings under iOS-Settings.
No change. Still one of the previous versions is loaded. 
Also tried to delete the app from IONIC View App, but after re-loading I am still getting the old one.

**Comment by gigocabrera** [comment]
**UPDATE:** the characters that are being added to my index.html page are ZERO WIDTH NO-BREAK SPACE. No visible space. Used to prevent a line break, as for example, after the hyphen in a word that contains its own hyphen.

    &#xFEFF;

Could this be related to encoding and line endings in Visual Studio 2013?

Besides this extra characters, the upload works fine for my on iOS

**Comment by zhangzhijiang** [comment]
@gigocabrera  I think you are right. I had the same issue. what I did to fix the issue was
1. remove &#xFEFF; from index.html
2. go to Visual Stduio 2013 File - Advanced Save Options
3. select Encoding - Unicode (UTF-8 without signature) -codepage 65001

**Comment by gigocabrera** [comment]
@zhangzhijiang you're the man!!! That worked like a charm, thank you!

**Comment by bjammin** [comment]
@zhangzhijiang Thanks for the fix, this also worked for mw.

**Comment by jacksonkr** [comment]
ionic upload was working just fine for me and now, for whatever reason, my current version is NOT being updated for ios nor android. This is very bad because a friend is paying me for the work and this is how they are getting updates. A number of friends are looking at this through the same app id so it's a hassle to have to give them a new ID for every update so how can I possibly get this to work as it just was? 

ionic info:
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-rc.2
Ionic CLI Version: 1.4.5
Ionic App Lib Version: 0.0.22
ios-deploy version: Not installed
ios-sim version: 3.1.1 
OS: Mac OS X Yosemite
Node Version: v0.12.1
Xcode version: Xcode 6.3.1 Build version 6D1002

**Comment by gigocabrera** [comment]
I'm experiencing the same issue. For me, ionic upload was working fine until yesterday. Now I see a `SIZE: NAN UNDEFINED` in ionic view and a `VIEW ERROR: STATUS 404` when I try to view the app

Any ideas?

**Comment by mlynch** [comment]
Did you try again this morning? There was a bad CLI release done yesterday

On Wed, Jun 3, 2015 at 8:57 AM, Luis Cabrera <notifications@github.com>
wrote:

> I'm experiencing the same issue. For me, ionic upload was working fine
> until yesterday. Now I see a SIZE: NAN UNDEFINED in ionic view and a VIEW
> ERROR: STATUS 404 when I try to view the app
>
> Any ideas?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/driftyco/ionic-view-issues/issues/10#issuecomment-108433148>
> .
>



-- 
Max Lynch
CEO and Co-founder
http://ionicframework.com/

**Comment by gigocabrera** [comment]
@mlynch thanks for the quick reply. I tried it several times this morning and even tried it again just now. I also tried removing the app_id from the ionic.project file to no avail. I haven't upgraded to the latest CLI. Could that be the issue?

**Comment by gigocabrera** [comment]
Here's what I have
ionic info:
 Cordova CLI: 5.0.0
 Gulp version:  CLI version 3.8.11
 Ionic Version: 1.0.0-rc.5
 Ionic CLI Version: 1.4.5
 Ionic App Lib Version: 0.0.22
 OS: Windows 8.1
 Node Version: v0.12.3

**Comment by jacksonkr** [comment]
@gigocabrera I decided I was going to have to removed my project particulars to get this to work and send out all new invites, and that worked. Below are the steps I took.

- renamed `ionic.project` to `ionic.project.bak`
- $ ionic login (followed by my credentials
- $ ionic upload
- $ ionic share some@email.net

Viola. Side note, you'll see that this gives you a new app id as well. GL.

**Comment by gigocabrera** [comment]
@jacksonkr I tried your workaround but it didn't work for me. I even changed the name of the of the app in ionic.project but nothing. I keep seeing `SIZE: NAN UNDEFINED`

Is the size of the app displayed in apps.ionic.io?

**Comment by jacksonkr** [comment]
@gigocabrera Woah, I definitely did not see that. It sounds like the root of our issues are different :(

**Comment by gigocabrera** [comment]
Upgrading to the new onic CLI Version: 1.5.0 solved my issue. Kudos to the ionic team. Great stuff!!!

**Comment by jfkoch** [comment]
I"m still having issues with 1.5.0. It's an old app that was built before this was fixed. Would that have anything to do with it?

**Comment by Avareto** [comment]
Hi IONIC team,
I am facing a new, very strange problem: 
This minimalistic and very simple app (http://codepen.io/anon/pen/gpRejQ) is correctly displayed in every browser! 
But when I upload it to Ionic View it shows a blank page only. 
I worked on it for hours to find the problem but now I need your help.
Whats wrong with it?

Btw, deployment works again for me. Thanks!

**Comment by bjammin** [comment]
If I had to guess I'd say your problem is this:
<script src="//code.ionicframework.com/nightly/js/ionic.bundle.js"></script>

You should really download the script (and css file) and reference them locally.  

Specifically, the // at the start means "the current protocol" (either http:// or https://) but when running on the device, the current protocol is file:// so the files cannot be loaded from the internet.

**Comment by Avareto** [comment]
Your are right!. Thanks!

**Comment by jfbloom22** [comment]
I am also having issues with ionic upload.  Updating images in particular seem to not get updated in the ionic view app.  

Cordova CLI: 5.1.1
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.0
Ionic Version: 1.0.1
Ionic CLI Version: 1.6.1
Ionic App Lib Version: 0.3.3
ios-deploy version: 1.7.0 
ios-sim version: 4.1.1 
OS: Mac OS X Yosemite
Node Version: v0.12.1
Xcode version: Xcode 6.4 Build version 6E35b

**Comment by VictorCoding** [comment]
Hey guys would these be related to this issue?

http://stackoverflow.com/questions/31610365/ionic-deploy-ionicdeploy-load-not-working

**Comment by mikeLspohn** [comment]
I seem to be having this issue today. Just update to the newest cli 1.6.3 Any ideas for how to at least temporarily fix this?

**Comment by mikeLspohn** [comment]
What is the best workaround for now to get this issue temporarily resolved? I've tried renaming/deleting the ionic.project and the .io-config.js and had success once with that, but now that's not working for me either.

**Comment by chedched** [comment]
Same issue here.

**Comment by chedched** [comment]
Appears as if the problem just occurs with apps that have been created before the update. @jacksonkr 's solution worked for me.

**Comment by mikeLspohn** [comment]
That solution did not seem to work for me. Android 5.1.1

**Comment by rhacker** [comment]
I still have the same problem. 
```
Cordova CLI: 5.3.1
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.0
Ionic Version: 1.0.0-beta.13
Ionic CLI Version: 1.6.5
Ionic App Lib Version: 0.3.9
ios-deploy version: Not installed
ios-sim version: 5.0.1 
OS: Mac OS X Yosemite
Node Version: v4.1.1
Xcode version: Xcode 6.4 Build version 6E35b
```

**Comment by jwbuitenhuis** [comment]
I've tried restarting Ionic View and re-uploading and I keep getting the same outdated app version in Ionic View. I'd rather not change the app_id as I have people testing the app and giving them a new ID all the time would be non-ideal. (Ionic View app is a great idea!)

I'm also using Ionic Deploy, could that have impact?

**Comment by alejandromagnorsky** [comment]
In my case it updates my app in Ionic View for iOS after removing and reinstalling Ionic View. Clear data and Sync to latest doesn't seem to work as expected. Using iPhone 6 and iOS9

**Comment by athanph** [comment]
Upgrading from 1.7.10 to 1.7.11 worked for me. It updated not only the assets but the HTML pages as well. Synced to latest afterwards.

**Comment by sugir93** [comment]
hi, most of my functionality is not working in the playstore version of the app.
eg i have an auth0 fb login, its working fine in the development version of the app, but if build the app for play-store release from that point its not working,
Do any one have any guess, why its not working?

thanks in advance.

**Comment by tlancina** [comment]
Hey all, I'm going to close this as the original caching issue was resolved. For people using deploy please note that deploy requires the [Ionic Deploy plugin](https://github.com/driftyco/ionic-plugin-deploy) which isn't supported by View at this time, so it won't work.

If you believe you're still running into caching issues, please open a new issue with steps on how to reproduce and we can take a look, thanks!
